src/zkevm_specs/evm/execution/call_staticcall.py

```diff
@@ -25,13 +29,17 @@ def call(instruction: Instruction):
     is_static = instruction.call_context_lookup(CallContextFieldTag.IsStatic)
     depth = instruction.call_context_lookup(CallContextFieldTag.Depth)
 
+    # Verify is_static == 1 for opcode STATICCALL.
+    instruction.constrain_zero((1 - is_call) * (1 - is_static.expr()))
```

Comment:
if is_call =1 & is_static=1 also pass this constraint I think, while they are invalid .

Reply:
It seems that the flag `is_static` is used to identify no modification (value == 0), as the [previous code](https://github.com/privacy-scaling-explorations/zkevm-specs/blob/master/src/zkevm_specs/evm/execution/call.py#L77) of opcode `CALL` as:
```
# Check not is_static if call has value
has_value = 1 - instruction.is_zero(value)
instruction.constrain_zero(has_value * is_static)
```
It seems that `is_static` could also be `1` if `value == 0` for opcode `CALL`.

Reply:
here I mean `is_call` and `is_static` can not be true at the same time, because one represents  `call` op code and the other represents `static_call` op code. not about call value matter .

Reply:
seems constrain` is_static = 1 -  is_call` makes sense, thus these two values can't be identical anyway and only one can be true.